lib/kubeclient.rb

```diff
@@ -355,20 +353,22 @@ def create_faraday_client(url = nil)
       }
 
       Faraday.new(url, options) do |connection|
-        if @auth_options[:username] && @auth_options[:password]
+        if @auth_options[:username]
           connection.basic_auth(@auth_options[:username], @auth_options[:password])
+        elsif @auth_options[:bearer_token]
+          connection.authorization(:Bearer, @auth_options[:bearer_token])
```

Comment:
Nice!  But this is printing warnings.  Actually `basic_auth` also does now.

> WARNING: `Faraday::Connection#authorization` is deprecated; it will be removed in version 2.0.  
> While initializing your connection, use `#request(:authorization, ...)` instead.  
> See https://lostisland.github.io/faraday/middleware/authentication for more usage info.  
> WARNING: `Faraday::Connection#basic_auth` is deprecated; it will be removed in version 2.0.  
> While initializing your connection, use `#request(:basic_auth, ...)` instead.  
> See https://lostisland.github.io/faraday/middleware/authentication for more usage info.  
(deprecation was here: https://github.com/lostisland/faraday/pull/1306)

I think it's supposed to be (untested):
```suggestion
          connection.request(:basic_auth, @auth_options[:username], @auth_options[:password])
        elsif @auth_options[:bearer_token]
          connection.request(:authorization, 'Bearer', @auth_options[:bearer_token])
```